projects/public-search/src/app/routes/base-route.ts

```diff
@@ -0,0 +1,103 @@
+/*
+ * RERO ILS UI
+ * Copyright (C) 2020 RERO
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, version 3 of the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { UrlSegment } from '@angular/router';
+import { TranslateService } from '@ngx-translate/core';
+import { Observable, Subscriber } from 'rxjs';
+
+export class BaseRoute {
+
+  private _translateService: TranslateService;
+
+  /**
+   * Constructor
+   * @param _translateService - TranslateService
+   */
+  protected constructor(
+    translateService: TranslateService
+  ) {
+    this._translateService = translateService;
+  }
+
+  /**
+   * Route matcher
+   * @param url - any
+   * @param type - string
```

Comment:
move `type` to `types` and change doc string

Reply:
done